### PR TITLE
CFY-3034 - Implementation of snapshots REST API.

### PR DESCRIPTION
**Pull request by simon-bar**

### CODE REVIEW

File: cloudify_rest_client/snapshots.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
+########
+# Copyright (c) 2014 GigaSpaces Technologies Ltd. All rights reserved
```
Comment (idanmo) [inline cloudify_rest_client/snapshots.py]: 2015

File: cloudify_rest_client/snapshots.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+#    * limitations under the License.
+
+import os
+import urlparse
+import contextlib
+
+from cloudify_rest_client import bytes_stream_utils
+
+
+class Snapshot(dict):
```
Comment (idanmo) [inline cloudify_rest_client/snapshots.py]: Please add doc strings for all public properties, methods and classes.

File: cloudify_rest_client/snapshots.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    """
+
+    def __init__(self, api):
+        self.api = api
+
+    def list(self, _include=None):
+        """
+        Returns a list of currently stored snapshots.
+
+        :param _include: List of fields to include in responce.
```
Comment (simon-bar) [inline cloudify_rest_client/snapshots.py]: responSe

File: cloudify_rest_client/snapshots.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        :return: Snapshots list.
+        """
+        response = self.api.get('/snapshots', _include=_include)
+        return [Snapshot(item) for item in response]
+
+    def create(self, snapshot_id):
+        """
+        Creates a new snapshot.
+
+        :param snapshot_id: Snapshot id of the new created deployment.
```
Comment (simon-bar) [inline cloudify_rest_client/snapshots.py]: of the snapshot to be created

File: cloudify_rest_client/snapshots.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+    def upload(self, snapshot_path, snapshot_id):
+        """
+        Uploads snapshot archive to Cloudify's manager.
+
+        :param snapshot_path: Path to snapshot archive.
+        :param snapshot_id: Id of the uploaded snapshot.
+        :return: Uploaded snapshot.
+
+        Snapshot archive should be this which was created and downloaded
```
Comment (simon-bar) [inline cloudify_rest_client/snapshots.py]: should be the same file that had been created and downloaded

File: cloudify_rest_client/snapshots.py [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+        from Cloudify's manager in process of create snapshot / download
```
Comment (simon-bar) [inline cloudify_rest_client/snapshots.py]: from Cloudify's manager as a result of create snapshot / ...

**Pull request closed without merging**